client/hwlib/src/helpers.rs

```diff
@@ -0,0 +1,78 @@
+/* Copyright 2024 Canonical Ltd.
+ * All rights reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Written by:
+ *        Nadzeya Hutsko <nadzeya.hutsko@canonical.com>
+ */
+
+#[cfg(test)]
+pub(crate) mod test_utils {
+    use anyhow::{bail, Result};
+    use once_cell::sync::Lazy;
+    use std::env;
+    use std::fs::read_dir;
+    use std::path::PathBuf;
+
+    pub(crate) fn get_test_filepath(file_name: &str) -> &'static str {
+        fn build_test_filepath(file_name: &str) -> String {
+            let mut path = get_project_root().unwrap();
+            if !path.to_str().unwrap().contains("hwlib") {
+                // If hwlib is not in the path, the project root is the monorepo root
+                path.push("client");
+                path.push("hwlib");
+            }
+            path.push("tests");
+            path.push("test_data");
+            path.push(file_name);
```

Comment:
I think it should be possible to clean up these repeated `.push` calls by using `.extend([...])`

Reply:
85a3538